File: Expense Tool Web Api/Domain/Entities/Expense.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Expense_Tool_Web_Api.Domain.Entities
{
    public class Expense
    {
        public int Id { get; set; }

        [Required]
        public DateTime Date { get; set; }

        [Required]
        [MaxLength(50)]
        public string? ExpenseType { get; set; }

        [Required]
        public string? Description { get; set; }

        [Required]
        [Range(0.00, double.MaxValue)]
        public decimal Amount { get; set; }

        [Required]
        [MaxLength(3)]
        public string? Currency { get; set; }

        public DateTime CreatedAt { get; set; }
        public DateTime LastUpdated { get; set; }
    }
}

File: Expense Tool React Frontend/src/interfaces/expense.ts
export interface Expense {
  id: number;
  date: string;
  expenseType: string;
  description: string;
  amount: number;
  currency: string;
  createdAt: Date;
  lastUpdated: Date;
}

export interface CreateExpenseDto {
  date: string;
  expenseType: string;
  description: string;
  amount: number;
  currency: string;
}

export interface ExpenseListDto {
  items: Expense[];
  pageIndex: number;
  totalPages: number;
  totalCount: number;
  hasPreviousPage: boolean;
  hasNextPage: boolean;
}
